src/envoy/mixer/config.cc

```diff
@@ -64,31 +59,54 @@ void ReadStringMap(const Json::Object& json, const std::string& name,
 }  // namespace
 
 void MixerConfig::Load(const Json::Object& json) {
-  ReadStringMap(json, kMixerAttributes, &mixer_attributes);
-  ReadStringMap(json, kForwardAttributes, &forward_attributes);
+  ReadStringMap(json, kMixerAttributes, http_config.mutable_mixer_attributes());
+  ReadStringMap(json, kForwardAttributes,
+                http_config.mutable_forward_attributes());
 
-  ReadString(json, kQuotaName, &quota_name);
-  ReadString(json, kQuotaAmount, &quota_amount);
+  AttributesBuilder builder(http_config.mutable_mixer_attributes());
+  if (json.hasObject(kQuotaName)) {
+    builder.AddString("quota.name", json.getString(kQuotaName));
+  }
+  if (json.hasObject(kQuotaAmount)) {
+    builder.AddInt64("quota.amount", std::stoi(json.getString(kQuotaAmount)));
+  }
+
+  // Copy mixer_attributes to TCP config.
+  *tcp_config.mutable_mixer_attributes() = http_config.mixer_attributes();
+
+  TransportConfig trans_config;
+  // Default is open, unless it specifically set to "close"
+  trans_config.set_network_fail_policy(TransportConfig::FAIL_OPEN);
+  if (json.hasObject(kNetworkFailPolicy) &&
+      json.getString(kNetworkFailPolicy) == "close") {
+    trans_config.set_network_fail_policy(TransportConfig::FAIL_CLOSE);
+  }
 
-  ReadString(json, kNetworkFailPolicy, &network_fail_policy);
+  trans_config.set_disable_check_cache(
+      json.getBoolean(kDisableCheckCache, false));
+  trans_config.set_disable_quota_cache(
+      json.getBoolean(kDisableQuotaCache, false));
+  trans_config.set_disable_report_batch(
+      json.getBoolean(kDisableReportBatch, false));
 
-  disable_check_cache = json.getBoolean(kDisableCheckCache, false);
-  disable_quota_cache = json.getBoolean(kDisableQuotaCache, false);
-  disable_report_batch = json.getBoolean(kDisableReportBatch, false);
+  // Copy to http and tcp
+  *http_config.mutable_transport() = trans_config;
+  *tcp_config.mutable_transport() = trans_config;
 
-  disable_tcp_check_calls = json.getBoolean(kDisableTcpCheckCalls, false);
+  tcp_config.set_disable_check_calls(
+      json.getBoolean(kDisableTcpCheckCalls, false));
 }
 
-void MixerConfig::ExtractQuotaAttributes(Attributes* attr) const {
-  if (!quota_name.empty()) {
-    int64_t amount = 1;  // default amount to 1.
-    if (!quota_amount.empty()) {
-      amount = std::stoi(quota_amount);
-    }
+void MixerConfig::CreateLegacyConfig(
```

Comment:
s/CreateLegacyConfig/CreateLegacyRouteConfig? Otherwise I was expecting this to refer to the legach top-level config.

Reply:
Will do